services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/team/MembershipRequestManagerImpl.java

```diff
@@ -113,6 +113,10 @@ public MembershipRequest create(UserInfo userInfo,
 			throw new IllegalArgumentException("This team is already open for the public to join, membership requests are not needed.");
 		}
 
+		if (team.getCanRequestMembership() != null && !team.getCanRequestMembership()) {
```

Comment:
Consider adding an enumeration (that is not exposed in the API), that translates the state of the two booleans into the state of the team E.g. (OPEN, PUBLIC, CLOSED) that is saved to the DB. Then instead of checking the booleans in the business logic we can work at the status level. Note that this status property also need to be migrated (but first we need to make sure we migrate the new requestMembership boolean).